internal/infrastructure/kubernetes/infra_test.go

```diff
@@ -12,10 +12,17 @@ import (
 	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
 
 	"github.com/envoyproxy/gateway/internal/envoygateway"
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
 	"github.com/envoyproxy/gateway/internal/ir"
 )
 
 func TestCreateInfra(t *testing.T) {
+	expected := ir.NewInfra()
+	// Apply the expected labels to the proxy infra.
+	expected.GetProxyInfra().GetProxyMetadata().Labels = envoyLabels()
```

Comment:
based on https://github.com/envoyproxy/gateway/issues/173#issuecomment-1238663746 I was under the assumption the metadata would be added as annotations. Can you please elaborate on why that changed ?

Reply:
nvm, based on the current logic the labels are used to setup a watch in the gateway reconciler

Reply:
From the [Kube Annotations/Labels docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#:~:text=Labels%20can%20be%20used%20to,characters%20not%20permitted%20by%20labels.):

>Labels can be used to select objects and to find collections of objects that satisfy certain conditions. In contrast, annotations are not used to identify and select objects. The metadata in an annotation can be small or large, structured or unstructured, and can include characters not permitted by labels.

We are using the labels to __select objects__, e.g. select the Gateway to reconcile from the gateway controller Service watch. The PR's label-based approach appears to be consistent with the Kube docs.

@youngnick can you elaborate on why you believe Annotations should be used instead of Labels?

Reply:
@arkodg since this PR is blocking several other PRs, I created https://github.com/envoyproxy/gateway/issues/368 to resolve this issue. In the meantime, I would like to proceed with labels.